test/validator/deprecated_fields_test.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test/test.dart';
+
+import 'package:pub/src/entrypoint.dart';
+import 'package:pub/src/validator.dart';
+import 'package:pub/src/validator/deprecated_fields.dart';
+
+import '../descriptor.dart' as d;
+import '../test_pub.dart';
+import 'utils.dart';
+
+Validator deprecatedFields(Entrypoint entrypoint) =>
+    new DeprecatedFieldsValidator(entrypoint);
+
+main() {
+  setUp(d.validPackage.create);
+
```

Comment:
It would be good to add a test that this doesn't warn for a pubspec that doesn't have either deprecated field.

Reply:
Done